gpflow/conditionals.py

```diff
@@ -181,13 +183,15 @@ def _sample_conditional(Xnew, X, kern, f, *, q_sqrt=None, white=False, full_cov=
     samples = _sample_mvn(mean, cov, cov_structure, num_samples=num_samples)
     if full_cov:
         samples = tf.matrix_transpose(samples)
+        mean = tf.matrix_transpose(mean)
```

Comment:
Hm, was this code broken before (in the full_cov=True case) ? What shape is mean supposed to be - could we sort that out in this PR? Also, would be good to bring this in line with the other _sample_conditional: introduce a local variable mean_PN that is the potentially transposed one so we don't have to transpose twice for the return value.